### PR TITLE
move plugin_management into core

### PR DESCRIPTION
**Pull request by mschrimpf**

### CODE REVIEW

**Comment by mschrimpf** [comment]
All tests pass for me locally
image

**Review by kvfairchild — COMMENTED** [review]
If we're going the persistent file route, are we sure that we want three separate testing directories, vs. starting with e.g. a dummy library that contains a dummy plugin that can be minimally modified as needed in test setup? It's easier to spot inconsistencies in behavior, etc. with one as opposed to maintaining multiples.

File: .travis.yml [hunk 1]
```diff
@@ -9,4 +9,4 @@ install:
   - python -m pip install -e ".[test]"
   - pip list
 script:
-  - pytest -m "not requires_gpu and not memory_intense and not slow and not travis_slow"
+  - pytest -m "not requires_gpu and not memory_intense and not slow and not travis_slow and not requires_conda"
```
[the remark this answers is not in the document]
Comment (mschrimpf) [reply .travis.yml:12]: travis does not have conda installed it seems: https://app.travis-ci.com/github/brain-score/core/jobs/588711648
image

File: brainscore_core/plugin_management/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# A(n incomplete) guide to plugins
```
[the remark this answers is not in the document]
Comment (mschrimpf) [reply brainscore_core/plugin_management/README.md:1]: should we then integrate the BSL doc here for a general-purpose plugin readme?

**Comment by mschrimpf** [comment]
> If we're going the persistent file route, are we sure that we want three separate dummy testing directories, vs. starting with e.g. a dummy library that contains a dummy plugin that can be minimally modified as needed in test setup? It's easier to spot inconsistencies in behavior, etc. with one as opposed to maintaining multiples.

The reason I had three separate repositories was that they all needed slightly different things, so I wanted to avoid a potentially growing mega directory that will have to cater to all sorts of different tests. I'm not sure how you would setup the "base" dummy repository, can you post an example?

File: .travis.yml [hunk 3]
```diff
@@ -9,4 +9,4 @@ install:
   - python -m pip install -e ".[test]"
```
Comment (kvfairchild) [inline .travis.yml:9]: For `conda` testing

```suggestion
  - python -m pip install -e ".[test]"
  # install conda for plugin management
  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
  - bash miniconda.sh -b -p $HOME/miniconda
  - source "$HOME/miniconda/etc/profile.d/conda.sh"
  - hash -r
  - conda config --set always_yes yes --set changeps1 no
  - conda update -q conda
  - conda info -a
```

File: brainscore_core/plugin_management/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kvfairchild) [reply brainscore_core/plugin_management/README.md:1]: And link to it from BSL/BSV? That sounds like the right approach.

[the remark this answers is not in the document]
Comment (mschrimpf) [reply brainscore_core/plugin_management/README.md:1]: Which part of the BSL documentation are you actually referring to? I can't find other detailed explanations of plugins

[the remark this answers is not in the document]
Comment (mschrimpf) [reply brainscore_core/plugin_management/README.md:1]: Yes agree; I would have moved the tutorials to `core` and linked from `language` (and later `vision`)

**Review by kvfairchild — APPROVED** [review]

**Pull request merged**